DigitalLearningSolutions.Web/Scripts/trackingSystem/centreCourseSetup.ts

```diff
@@ -20,21 +20,26 @@ function copyLaunchCourseLinkToClipboard(customisationId: string) {
   const launchCourseButtonId = launchCourseButtonIdPrefix + customisationId;
   const launchCourseButton = document.getElementById(launchCourseButtonId) as HTMLAnchorElement;
   const link = launchCourseButton.href;
-  const succeeded = copyTextToClipboard(link);
-  const alertMessage = succeeded
-    ? `Copied the text: ${link}`
-    : `Copy not supported or blocked. Try manually selecting and copying: ${link}`;
-
-  alert(alertMessage);
+  copyTextToClipboard(link);
 }
 
-function copyTextToClipboard(textToCopy: string): boolean {
-  try {
-    navigator.clipboard.writeText(textToCopy);
-    return true;
-  } catch (e) {
-    return copyTextToClipboardFallback(textToCopy);
+function copyTextToClipboard(textToCopy: string): void {
+  if (!navigator.clipboard) {
+    const succeeded = copyTextToClipboardFallback(textToCopy);
+    if (succeeded) {
+      displaySuccessAlert(textToCopy);
+    } else {
+      displayFailureAlert(textToCopy);
+    }
+    return;
   }
+
+  navigator.clipboard.writeText(textToCopy).then(() => {
+    displaySuccessAlert(textToCopy);
+  },
+  () => {
+    displayFailureAlert(textToCopy);
+  });

```

Comment:
That's odd.. I'd expect Promise syntax to be something like:
```
navigator.clipboard.writeText(textToCopy)
    .then(() => displaySuccessAlert(textToCopy))
    .catch(() => displayFailureAlert(textToCopy));
```

Also, I'm assuming clipboard doesn't work on IE and is just replaces with fallback?

Reply:
First example I saw for Promise was as it was previously written, but your example looks neater so I've used that instead. And you are correct, IE has to use the fallback as navigator.clipboard doesn't exist in IE.